.changeset/wet-foxes-walk.md

```diff
@@ -0,0 +1,28 @@
+---
+'astro': major
+---
+
+`Astro.params` aren't decoded anymore. 
+
+### [changed]: `Astro.params` aren't decoded anymore.
+In Astro v4.x, `Astro.params` were decoded using `decodeURIComponent`. 
+
+Astro v5.0 doesn't decode `Astro.params` anymore, so you'll need to manually decode them yourself.
+
+#### What should I do?
+If you were relying on `Astro.params` being decoded for you, you'll need to manually decode it using `decodeURI`.
+
+The use of [`decodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent)) is discouraged because it decodes more characters than it should, for example `/`, `?`, `#` and more.
+
+```diff
+---
+export function getStaticPaths() {
+  return [
++    { params: { id: decodeURI("%5Bpage%5D") } },
+-    { params: { id: "%5Bpage%5D" } },
+  ]
+}
+
+const { id } = Astro.params;
+---
+```
```

Comment:
Maybe we can re-phrase it to something like this, because I think if we say `Astro.params` isn't decoded anymore, it may return encoded strings, which isn't accurate. Or maybe I'm confused with `Astro.params`?

`````suggestion
`params` passed in `getStaticPaths` are no longer automatically decoded.

### [changed]: `Astro.params` aren't decoded anymore.
In Astro v4.x, `params` in ` were automatically decoded using `decodeURIComponent`. 

Astro v5.0 doesn't automatically decode `params` in `getStaticPaths` anymore, so you'll need to manually decode them yourself if needed

#### What should I do?
If you were relying on the automatic decode, you'll need to manually decode it using `decodeURI`.

Note that the use of [`decodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent)) is discouraged for `getStaticPaths` because it decodes more characters than it should, for example `/`, `?`, `#` and more.

```diff
---
export function getStaticPaths() {
  return [
+    { params: { id: decodeURI("%5Bpage%5D") } },
-    { params: { id: "%5Bpage%5D" } },
  ]
}

const { id } = Astro.params;
---
```
`````